gazebo_plugins/test/test_gazebo_ros_wheelslip.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gazebo/test/ServerFixture.hh>
+#include <gazebo_ros/node.hpp>
+#include <gazebo_ros/testing_utils.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace std::chrono_literals;
+
+/// Tests the gazebo_ros_wheelslip plugin
+class GazeboRosWheelSlipTest : public gazebo::ServerFixture
+{
+};
+
+TEST_F(GazeboRosWheelSlipTest, SetSlipCompliance)
+{
+  // Load test world and start paused
+  this->Load("worlds/gazebo_ros_wheelslip.world", true);
+
+  // World
+  auto world = gazebo::physics::get_world();
+  ASSERT_NE(nullptr, world);
+
+  // Create node
+  auto node = std::make_shared<rclcpp::Node>("test_gazebo_ros_wheelslip");
+  ASSERT_NE(nullptr, node);
+
+  auto parameters_client_rear = std::make_shared<rclcpp::SyncParametersClient>(
+    node,
+    "trisphere_cycle_slip0/wheel_slip_rear");
+
+  std::map<std::string, double> parameters_rear_pairs = {
+    {"slip_compliance_unitless_lateral", 5},
+    {"slip_compliance_unitless_lateral/wheel_rear_left", 0.0},
+    {"slip_compliance_unitless_lateral/wheel_rear_right", 5},
+    {"slip_compliance_unitless_longitudinal", 6},
+    {"slip_compliance_unitless_longitudinal/wheel_rear_left", 4},
+    {"slip_compliance_unitless_longitudinal/wheel_rear_right", 6}};
+
+  int counter = 0;
+  while (!parameters_client_rear->wait_for_service(1s)) {
+    if (!rclcpp::ok()) {
+      rclcpp::shutdown();
+    }
```

Comment:
I'm confused by this logic. Maybe you meant to call `rclcpp::init()` instead? If this is necessary I would suggest doing it in the test fixture `setUpTestCase` method.